Glyssen/Rules/CharacterGroupGenerator.cs

```diff
@@ -914,13 +1021,118 @@ public static void DistributeBooksAmongNarratorGroups(List<AuthorStats> authorSt
 			{
-				if (ExtraBiblicalGroup == null)
+				if (group != null)
+				{
+					group.CharacterIds.Add(characterId);
 					return;
-				ExtraBiblicalGroup.CharacterIds.AddRange(includedCharacterDetails.
-					Select(cd => cd.CharacterId).Where(c => CharacterVerseData.IsCharacterStandard(c, false) &&
-					!Groups.Any(g => g.CharacterIds.Contains(c))));
+				}
+
+				// add to a narrator group
+				var narratorGroup = GetExtraBiblicalToNarratorGroup(characterId, characterType, narratorPrefix, dramatizationPreferences);
+
+				if (narratorGroup != null)
+					narratorGroup.CharacterIds.Add(characterId);
+			}
+
+			private CharacterGroup GetExtraBiblicalToNarratorGroup(string characterId, CharacterVerseData.StandardCharacter characterType,
+				string narratorPrefix, ProjectDramatizationPreferences dramatizationPreferences)
+			{
+				var dramatizationOption = ExtraBiblicalMaterialSpeakerOption.Omitted;
+
+				switch (characterType)
+				{
+					case CharacterVerseData.StandardCharacter.BookOrChapter:
+						dramatizationOption = dramatizationPreferences.BookTitleAndChapterDramatization;
+						break;
+
+					case CharacterVerseData.StandardCharacter.ExtraBiblical:
+						dramatizationOption = dramatizationPreferences.SectionHeadDramatization;
+						break;
+
+					case CharacterVerseData.StandardCharacter.Intro:
+						dramatizationOption = dramatizationPreferences.BookIntroductionsDramatization;
+						break;
+				}
+
+				if (dramatizationOption == ExtraBiblicalMaterialSpeakerOption.Omitted)
+					return null;
+
+				var narratorId = narratorPrefix + characterId.Split(new[] { '-' }, 2)[1];
+				List<CharacterGroup> possibleGroups;
+
+				if (dramatizationOption == ExtraBiblicalMaterialSpeakerOption.Narrator)
+					possibleGroups = NarratorGroups.Where(g => g.CharacterIds.Contains(narratorId)).ToList();
+				else
+					possibleGroups = NarratorGroups.Where(g => !g.CharacterIds.Contains(narratorId)).ToList();
+
+				if (possibleGroups.Count == 0)
+					possibleGroups = NarratorGroups.Except(possibleGroups).ToList();
+
+				CharacterGroup narratorGroup = null;
+
+				// this should never happen, so throw an exception if it does
+				if (possibleGroups.Count == 0)
+					throw new NullReferenceException(string.Format("No group was found for character '{0}'", characterId));
+
+				if (possibleGroups.Count == 1)
+				{
+					return possibleGroups[0];
+				}
+
+				// get the narrator group with the least amount of speaking
+				var leastEstimatedHours = double.MaxValue;
+				foreach (var cg in possibleGroups)
+				{
+					var estimatedHours = cg.EstimatedHours;
+
+					if (estimatedHours < leastEstimatedHours)
+					{
+						leastEstimatedHours = estimatedHours;
+						narratorGroup = cg;
+
+						// is this a good enough match?
+						if (leastEstimatedHours < 0.01)
+							break;
+					}
+				}
+
+				// this should never happen, so throw an exception if it does
+				if (narratorGroup == null)
```

Comment:
With about Linq expression, it should be obvious that this can't happen, so you can remove this check.